sirfilesystem.c

```diff
@@ -157,6 +157,75 @@ char* _sir_getcwd(void) {
 #endif
 }
 
+#if defined(__OpenBSD__)
+static int
+_sir_openbsdself(char* out, int capacity, int* dirname_length)
+{
+  char   buffer1[4096];
+  char   buffer2[PATH_MAX];
+  char   buffer3[PATH_MAX];
+  char** argv     = (char**)buffer1;
+  char*  resolved = NULL;
+  int    length   = -1;
+
+  for (;;) {
+    int mib[4] = { CTL_KERN, KERN_PROC_ARGS, getpid(), KERN_PROC_ARGV };
```

Comment:
So I take it the same mib that we use on FreeBSD doesn't jive on OpenBSD? And it has no readlink?

Reply:
Correct. They don't stash the executed path at all to query later.  And you can't use the readlink trick because they don't have proc.